_rules/auto-update-text-efbfc7.md

```diff
@@ -0,0 +1,389 @@
+rule_type: atomic
+description: |
+  This rule checks that there are mechanisms to pause, stop or hide the auto-updating of text content
+accessibility_requirements: # Remove whatever is not applicable
+  wcag20:2.2.2: # Pause, Stop, Hide (A)
+    forConformance: true
+    failed: not satisfied
+    passed: further testing needed
+    inapplicable: further testing needed
+input_aspects:
+  - DOM Tree
+  - CSS Styling
+acknowledgements:
+  authors:
+    - Carlos Duarte
+---
+
+## Applicability
+
+The rule applies to any [visible][] [text content][] in an [HTML document][] if:
+
+- the [text content][] is removed or replaced by new content that does not contain the previous content; and
+- the removal or replacing is not caused by a [user interaction][]; and
+- the removal or replacing happens anytime after the [readiness][document readiness] of the [HTML document][] the [text node][] belongs to is equal to "complete"; and
+- it is not the only [content][] in the [HTML document][].
+
+## Expectation
+
+For the test target a [user interface component][] is provided to pause, stop or hide the updating of the [text content][].
+
+**Note**: If there is more than one auto-updating [text content][], a single [user interface component][] may be used to pause, stop or hide updating all [text content][].
+
+## Assumptions
+
+This rule assumes that the auto-updating of the content is not [essential][], which is listed as valid exception to [Success Criterion 2.2.2: Pause, Stop, Hide][sc 2.2.2]. When the auto-updating of content is [essential][] this rule may produce incorrect results.
+
+## Accessibility Support
+
+_There are no major accessibility support issues known for this rule._
+
+## Background
+
+- [Understanding Success Criterion 2.2.2: Pause, Stop, Hide][sc 2.2.2]
+- [G186: Using a control in the Web page that stops moving, blinking, or auto-updating content][g186]
+- [F16: Failure of Success Criterion 2.2.2 due to including scrolling content where movement is not essential to the activity without also including a mechanism to pause and restart the content][f16]
+
+## Test Cases
+
+### Passed
+
+#### Passed Example 1
+
+The text content automatically updates every 3 seconds after the page completes loading. A button is available to stop the automatic updates.
+
+```html
+<body onload="start()">
+	<p>Random number: <span id="target">1</span></p>
+	<input type="button" onclick="stop()" value="Stop updates" />
+
+	<p>
+		The W3C Web Accessibility Initiative (WAI) develops standards and support materials to help you understand and
+		implement accessibility.
+	</p>
+
+	<script type="text/javascript">
+		function change() {
+			var target = document.getElementById('target')
+			var number = Math.floor(Math.random() * 1000)
+			target.innerText = number
+		}
+
+		var updates
+		function start() {
+			updates = setInterval(change, 3000)
+		}
+
+		function stop() {
+			clearInterval(updates)
+		}
+	</script>
```

Comment:
Most of the Javascript is shared between example. For the sake of separation of concerns, DRY, and keeping the examples easier to focus on the important changes, I suggest:
* Move Javascript to a `test-assets` file.
* Put in it all the code: `change`, `start`, `toggle`, `stop`, `hide` and the `updating` boolean.
* (maybe rename to `startUpdates` and so on, maybe not…)
* Adapt `start` to also change `updating`.
* Systematically call `onload=start()` which makes purpose obvious. (I know I've been the one saying to put `onload=toggle()` at some places 😁 refactoring and polishing code is a very iterative process…)
* Only change the content of `onclick` in the individual examples. That will allow readers to focus on the important change and immediately agree with the description.

(Note that you can also decrease the update interval to 1s or 0.5s to make the behaviour of the examples even more obvious when trying them in the browser)

Reply:
Refactored